lib/msal-common/src/response/ServerAuthorizationTokenResponse.ts

```diff
@@ -51,4 +51,15 @@ export type ServerAuthorizationTokenResponse = {
     trace_id?: string;
     correlation_id?: string;
     claims?: string;
+
+    /**
+     * Unique to Managed Identity
+     */
+    // success
+    client_id?: string;
```

Comment:
Since these properties are kind of separate from the above ones, can they be extracted into something like `ManagedIdentityResponse` and then called like `networkClient.sendPostRequestAsync<ManagedIdentityResponse>?

Reply:
I could, and that's how I originally did it, but then I'd have to change more code in msal-common, which is what I'm trying to avoid. I'd have two different server responses. In ResponseHandler on line 399, I'd have to change the type from ServerAuthorizationTokenResponse to ServerAuthorizationTokenResponse | ManagedIdentityTokenResponse, and then the rest of the code in that function would throw errors and I'd have to add a bunch of conditional checks to see if properties on the server response exist. Thoughts?

Reply:
Agree with Peter, this is violating the Single Responsability Principle. This object is only for ESTS responses. We should have a different response for MSI - MSIResponse and if needed, an adapter (e.g. method on ServerAuthorizationResponse) that converts between MSIResponse to ServerAuthorizationResponse.